org.eclipse.m2e.repository/category.xml

```diff
@@ -18,6 +18,18 @@
    <feature id="org.eclipse.m2e.sdk.feature">
       <category name="m2e"/>
    </feature>
+   <bundle id="biz.aQute.bnd.util">
+      <category name="m2e"/>
+   </bundle>
+   <bundle id="biz.aQute.bnd.util.source">
+      <category name="m2e"/>
+   </bundle>
+   <bundle id="biz.aQute.bndlib">
+      <category name="m2e"/>
+   </bundle>
+   <bundle id="biz.aQute.bndlib.source">
+      <category name="m2e"/>
+   </bundle>
```

Comment:
This should be a dedicated category like "Transitive dependencies" or, even better but harder to setup, no category at all (but removing a category create a dumb "Uncategorized" category....)

Reply:
as mentined above I don't see any value in adding this in the category as no one want to install this independent from the pde feature... it should simply be a Dependecy their with proper version range...

Reply:
Making them a dependency in the feature doesn't include them in the p2 repository. Things that are included in p2 repo are only what's explicitly part of the category.xml or .product files, and their *included* dependencies. Turning an inclusion into a requirement usually removes the content from the p2 repo, so we need to explicit somewhere else to add the content to the p2 repo.

Reply:
I don't think that this is true, is there an integration test showing that? At the p2 level there is no difference between an `included` versus `depended` they both end up as a (direct) requirement that is then included in the update-site (even if not visible in the install dialog).

Reply:
> I don't think that this is true, is there an integration test showing that?

Please try and see by yourself.

Reply:
OK, and now try using a version range instead of 1 specific version; you'll see the bundle is not added.
So for require/import to work, you need to hardcode the version, which is even less flexible than an inclusion. There is no benefit in doing it. This has been a long-known (and probably reported in some trackers) limitation. However, it is not clear what would be a better strategy. I looked up a bit as I cannot remember where is the right `if`, and I believe it's somewhere in p2 Mirroring class.

Reply:
> This should be a dedicated category like "Transitive dependencies" or, even better but harder to setup, no category at all (but removing a category create a dumb "Uncategorized" category....)

Good point! I will see what I can achieve.

Reply:
I have created https://github.com/eclipse/tycho/issues/898
Until this is resolved we should stick to `include` then, that's the version we have build, tested and veted our build and I don't see a value to take additional effort to move that around as it is a requirement of the pde feature and not of m2e in general.

Reply:
The benefit is that it doesn't require a rebuild of the feature to update the version.

Reply:
as the feature and the bundle are part of the same build I don't see an issue here and otherwise I see the risk that p2 won't update the version as nothing is changed (neither feature nor bundle).

Reply:
> This should be a dedicated category like "Transitive dependencies" or, even better but harder to setup, no category at all (but removing a category create a dumb "Uncategorized" category....)

I removed the `category` elements so that they are uncategorized and in my local test build they were included.

But I cannot judge how sever Christophs suspected risk is. But I can test locally if an update of an uncategorized bundle in the repo that is not included in any feature works or not. Of course that would only be for one case. Maybe there are other cases.